### PR TITLE
fix(sdk-lib-mpc): use rejection sampling for range proof challenge

### PR DESCRIPTION
**Pull request by zhongxishen**
TICKET: HSM-165

### CODE REVIEW

File: modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts [hunk 1]
```diff
@@ -194,6 +194,42 @@ export async function verifyNtildeProof(
   }
   return true;
 }
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (zahin-mohammad) [inline modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts:198]: Why is this needed?

File: modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts [hunk 2]
```diff
@@ -241,10 +277,16 @@ export async function prove(
   hash.update(bigIntToBufferBE(w, modulusBytes));
   hash.update('$');
   const e = bigIntFromBufferBE(hash.digest()) % q;
+  // Note: steps for migration
+  //     1. upgrade SDK to this, producing old challenge but ready to verify proof with both old and new challenge
+  //     2. upgrade HSM, producing new challenge and ready to verify proof with both old and new challenge
+  //     3. upgrade SDK to the next version that produces new challenge (possibly verify proof with only new challenge)
+  // const { nonce, e } = rangeProofChallenge(curve, modulusBytes, pk, c, z, u, w);
```
Comment (zahin-mohammad) [inline modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts:284]: So sounds like the prq-requisite to merging this SDK pr is to first have the HSM and WP ready to support both old/new challenge and challenge proof values.

Should this PR be a draft for now?

File: modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts [hunk 3]
```diff
@@ -241,10 +277,16 @@ export async function prove(
...
   const s = (modPow(r, e, pk.n) * beta) % pk.n;
   const s1 = e * m + alpha;
   const s2 = e * rho + gamma;
   return { z, u, w, s, s1, s2 };
+  // return { z, u, w, nonce, s, s1, s2 };
```
Comment (zahin-mohammad) [inline modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts:289]: Remove

File: modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts [hunk 4]
```diff
@@ -301,6 +355,56 @@ export function verify(
   return proof.w === products;
 }
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (zahin-mohammad) [inline modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts:358]: Do we still need this?

**Review by zahin-mohammad — COMMENTED** [review]
in general lgtm, but it seems we are not ready to merge this in yet. If thats the case please add the appropriate github label for future readers.

File: modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts (hunk 2 above)
Comment (zhongxishen) [reply modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts:284]: Well, this PR is ready for step 1 mentioned in the notes, upgrading to it shouldn't break anything but SDK will be ready for step 2.  The commented out code and disabling of eslint warnings are the things we need to enable in step 3.

Comment (zhongxishen) [reply modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts:284]: Agreed to make it draft for now, because a number of other changes that go with this proof in a single call to HSM are not settled.  See https://github.com/BitGo/hsm-firmware/pull/812/files#diff-d61bc8c0ab7f3b380b8be40aaac80278f55d32402ae71244dc774ab9754b6b8aR1263

**Review by zahin-mohammad — COMMENTED** [review]
(flushing from review Q)

**Comment by zhongxishen** [comment]
stop gg18 work
